### PR TITLE
BUG: mode not sorting values for arrow backed strings

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -2763,13 +2764,23 @@ def test_rolling_wrong_param_min_period():
...
 
 
-def test_by_column_values_with_same_starting_value():
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        object,
+        pytest.param(
+            "string[pyarrow_numpy]",
+            marks=pytest.mark.skipif(pa_version_under7p0, reason="arrow not installed"),
```
Comment (mroeschke) [inline pandas/tests/groupby/test_groupby.py:2773]: Could you use an `importorskip` in the test when `dtype` is `pyarrow_numpy`? (avoids having to change `pa_version_under7p0` every time we bump pyarrow)

Comment (phofl) [reply pandas/tests/groupby/test_groupby.py:2773]: thoughts about creating a pyarrow_installed variable?

I like this pattern a bit better than the if inside the test

Comment (mroeschke) [reply pandas/tests/groupby/test_groupby.py:2773]: > thoughts about creating a pyarrow_installed variable?

Sure this would be good too

Comment (phofl) [reply pandas/tests/groupby/test_groupby.py]: done

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -2763,13 +2764,23 @@ def test_rolling_wrong_param_min_period():
...
 
 
-def test_by_column_values_with_same_starting_value():
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        object,
+        pytest.param(
+            "string[pyarrow_numpy]",
+            marks=pytest.mark.skipif(pa_installed, reason="arrow not installed"),
```
Comment (lithomas1) [inline pandas/tests/groupby/test_groupby.py:2773]: This condition is backwards?

Comment (phofl) [reply pandas/tests/groupby/test_groupby.py]: oh thx, oversight

File: pandas/tests/groupby/test_groupby.py [hunk 3]
```diff
@@ -2541,7 +2542,10 @@ def test_groupby_column_index_name_lost(func):
     "infer_string",
     [
         False,
-        True,
+        pytest.param(
+            True,
+            marks=pytest.mark.skipif(not pa_installed, reason="arrow not installed"),
```
Comment (mroeschke) [inline pandas/tests/groupby/test_groupby.py:2547]: I just realized we have a `td.skip_if_no` mark as well

File: pandas/compat/pyarrow.py [hunk 4]
```diff
@@ -7,13 +7,19 @@
...
     pa_version_under11p0 = _palv < Version("11.0.0")
     pa_version_under12p0 = _palv < Version("12.0.0")
     pa_version_under13p0 = _palv < Version("13.0.0")
     pa_version_under14p0 = _palv < Version("14.0.0")
 except ImportError:
+    pa_installed = False
+    pa_version_under7p0 = True
+    pa_version_under8p0 = True
+    pa_version_under9p0 = True
+    pa_version_under10p0 = True
```
Comment (lithomas1) [inline pandas/compat/pyarrow.py:22]: Why do we need these?

File: pandas/tests/groupby/test_groupby.py (hunk 3 above)
Comment (phofl) [reply pandas/tests/groupby/test_groupby.py:2547]: does this work inside of pytest.param as well?

File: pandas/compat/pyarrow.py (hunk 4 above)
Comment (phofl) [reply pandas/compat/pyarrow.py]: got in while merging main, sorry

File: pandas/tests/groupby/test_groupby.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/groupby/test_groupby.py:2547]: It should, yeah. It returns a `skipif` marker

Comment (phofl) [reply pandas/tests/groupby/test_groupby.py]: Yep, thx

**Review by lithomas1 — APPROVED** [review]
Thanks, this LGTM.

I'll merge this tomorrow morning assuming no other comments.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**